docs/notebooks/annotated_mnist.ipynb

```diff
@@ -1,598 +1,655 @@
-    "  it to get a second key that you'll use for parameter initialization. (Learn\n",
-    "  more about\n",
-    "  [PRNG chains](https://flax.readthedocs.io/en/latest/design_notes/linen_design_principles.html#how-are-parameters-represented-and-how-do-we-handle-general-differentiable-algorithms-that-update-stateful-variables)\n",
-    "  and\n",
-    "  [JAX PRNG design](https://github.com/google/jax/blob/main/design_notes/prng.md).)"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 18,
-   "metadata": {
-    "id": "n53xh_B8Ht_W"
-   },
-   "outputs": [],
-   "source": [
-    "rng = jax.random.PRNGKey(0)\n",
-    "rng, init_rng = jax.random.split(rng)"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {
-    "id": "Y3iHFiAuH41s"
-   },
-   "source": [
-    "## 13. Initialize train state\n",
-    "\n",
-    "Remember that function initializes both the model parameters and the optimizer\n",
-    "and puts both into the training state dataclass that is returned."
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 19,
-   "metadata": {
-    "id": "Mj6OfdEEIU-o"
-   },
-   "outputs": [],
-   "source": [
-    "learning_rate = 0.1\n",
-    "momentum = 0.9"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 20,
-   "metadata": {
-    "id": "_87fL90dH-0Z"
-   },
-   "outputs": [],
-   "source": [
-    "state = create_train_state(init_rng, learning_rate, momentum)\n",
-    "del init_rng  # Must not be used anymore."
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {
-    "id": "UqNrWu7kIC9S"
-   },
-   "source": [
-    "## 14. Train and evaluate\n",
-    "\n",
-    "Once the training and testing is done after 10 epochs, the output should show that your model was able to achieve approximately 99% accuracy."
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 21,
-   "metadata": {
-    "id": "0nxgS5Z5IsT_"
-   },
-   "outputs": [],
-   "source": [
-    "num_epochs = 10\n",
-    "batch_size = 32"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 22,
-   "metadata": {
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "9q9JZjt16AMf"
+      },
+      "source": [
+        "# Annotated MNIST\n",
+        "\n",
+        "This tutorial demonstrates how to construct a simple convolutional neural\n",
+        "network (CNN) using the [Flax](https://flax.readthedocs.io) Linen API and train\n",
+        "the network for image classification on the MNIST dataset.\n",
+        "\n",
+        "Note: This notebook is based on Flax's official\n",
+        "[MNIST Example](https://github.com/google/flax/tree/main/examples/mnist).\n",
+        "If you see any changes between the two feel free to create a\n",
+        "[pull request](https://github.com/google/flax/compare)\n",
+        "to synchronize this Colab with the actual example."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Check GPU\n",
```

Comment:
This cell seems to be new, why did you add it?

Reply:
To make clear that the example runs on a Colab with GPU.

But looking at the RTD this is actually quite visible now ...
https://flax--2071.org.readthedocs.build/en/2071/notebooks/annotated_mnist.html

What would you prefer? Removing it or changing to `!nvidia-smi -L` that will have a sinle line output?

Reply:
Makes sense, I actually think it is quite nice it shows the GPU info, so I'm fine keeping it as is.